app/deck/file/openapi2kong.md

```diff
@@ -0,0 +1,94 @@
+---
+title: openapi2kong
+---
+
+The `openapi2kong` command converts an OpenAPI specification to Kong's declarative configuration format.
+
+* A Service is created, pointing to the URLs in the `servers` block
+* One route per `operationId` is created
+* If an `openIdConnect` securityScheme is present, an `openid-connect` plugin will be generated
+
+## Get started
+
+Converting an OpenAPI file to a Kong declarative configuration can be done in a single command:
+
+```bash
+deck file openapi2kong --spec oas.yaml --output-file kong.yaml
+```
+
+## Best practices
+
+### ID Generation
+
+The `openapi2kong` command will generate a declarative configuration file that contains an `id` for each entity. This ensures that even if the entity name (or any other identifying parameter) changes, it will update an entity in {{ site.base_gateway }} rather than deleting then recreating an entity.
+
+If you have an existing {{ site.base_gateway }} instance, you can use the `--inso-compatible` flag to skip ID generation. In this instance, decK will match entities on a best effort basis using the `name` field.
+
+If you have multiple teams using the same names and `operationId` values in their specifications, you can specify the `--uuid-base` flag to set a custom namespace when generating IDs.
+
+To learn more about ID generation, see the [openapi2kong GitHub documentation](https://github.com/Kong/go-apiops/blob/main/docs/oas2kong-id-generation-deck.md#id-generation).
+
+### Adding plugins
+
+`openapi2kong` allows you to configure plugins directly in your OpenAPI specification by providing an `x-kong-plugin-PLUGIN-NAME` annotation at the root level to add it to the service, at the path level to add it to all routes on that path, or on an operation to add it to a specific route.
+
+_However_, we do not recommend using this functionality. Routing and policy enforcement are generally owned by different teams, and you do not want to publish plugin configuration in an OpenAPI specification that is shared with customers.
```

Comment:
Should we even document this if we don't want people using it?

Reply:
Sadly it's used enough that if we don't document the functionality we'll get questions about it